ml/ml.cc

```diff
@@ -173,26 +176,26 @@ ml_features_preprocess(ml_features_t *features)
 */
 
 static void
-ml_kmeans_init(ml_kmeans_t *kmeans, size_t num_clusters, size_t max_iterations)
+ml_kmeans_init(ml_kmeans_t *kmeans)
 {
-    kmeans->num_clusters = num_clusters;
-    kmeans->max_iterations = max_iterations;
-
-    kmeans->cluster_centers.reserve(kmeans->num_clusters);
+    kmeans->cluster_centers.reserve(2);
```

Comment:
Is it not better to setup `num_clusters` as argument and set default value equals `2`? How did you arrive in the conclusion that we should use  `2`?

Reply:
Original idea here was that 2 is the most simple model that's more complex then 1. At some point I think we will make it into a param but just have not for now. 

Once models all being saved and loaded from db we will be looking again at default params and maybe as part of that we might as well expose this as a param to tinker with a bit, maybe allowing it to be 1 to 5 perhaps. 

Long long term it would be useful to be able to set all the params for specific models and then this param for sure would be important, eg I see a few dims at moment that often have bad models and for them k=3 would probably fix it. 

So agree makes sense to be a param at some stage when we get to it. 

For now to begin with we sort of wanted to keep it as simple as possible.